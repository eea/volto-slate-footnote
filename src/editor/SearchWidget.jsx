import React, { useState } from 'react';
import { Search, Card, Segment } from 'semantic-ui-react';
import { escapeRegExp, filter } from 'lodash';

const SearchWidget = (props) => {
  const [results, setResults] = useState([]);
  const { onChange, choices, value } = props;

  /**
   * evaluate on Regex to filter results
   * @param {Object} e - event
   * @param {Object} data
   */
  const handleSearchChange = (e, data) => {
    const re = new RegExp(escapeRegExp(data.value), 'i');
    const isMatch = (result) => re.test(result.footnote);
    const resultsFiltered = filter(choices, isMatch);

    setResults(resultsFiltered);
    onChange({ footnote: data.value });
  };

  /**
   * callback onChange for selection from results
   * @param {Object} e - event
   * @param {Object} data
   */
  const handleSelected = (e, data) => {
    const { footnote } = data.result;

    onChange({ footnote });
  };

  return (
    <div id={`blockform-fieldset-default`}>
      <Segment className="attached" fluid>
        <Card fluid>
          <Card.Content fluid>
            <Card.Header>Citation</Card.Header>
            <Card.Description fluid>
              <Search
                id="field-footnote"
                input={{ icon: 'search', iconPosition: 'right', fluid: true }}
                fluid
                onFocus={handleSearchChange}
                onResultSelect={handleSelected}
                onSearchChange={handleSearchChange}
                results={results}
                value={value}
              />
            </Card.Description>
          </Card.Content>
        </Card>
      </Segment>
    </div>
  );
};

export default SearchWidget;
