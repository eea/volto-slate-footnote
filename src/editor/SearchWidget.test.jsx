import React from 'react';
import renderer from 'react-test-renderer';
import SearchWidget from './SearchWidget';
describe('Test SearchWidget', () => {
  it('check html content', () => {
    const component = renderer.create(<SearchWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
