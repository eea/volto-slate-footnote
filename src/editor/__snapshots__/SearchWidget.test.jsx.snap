// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchWidget check html content 1`] = `
<div
  id="blockform-fieldset-default"
>
  <div
    className="ui segment attached"
    fluid={true}
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <div
        className="content"
        fluid={true}
      >
        <div
          className="header"
        >
          Citation
        </div>
        <div
          className="description"
          fluid={true}
        >
          <div
            className="ui fluid search"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="ui fluid right icon input"
            >
              <input
                autoComplete="off"
                className="prompt"
                id="field-footnote"
                onChange={[Function]}
                onClick={[Function]}
                tabIndex="0"
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                className="search icon"
                onClick={[Function]}
              />
            </div>
            <div
              className="results transition"
            >
              <div
                className="message empty"
              >
                <div
                  className="header"
                >
                  No results found.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
